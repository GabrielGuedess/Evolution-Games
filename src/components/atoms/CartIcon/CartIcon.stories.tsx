import { Story, Meta } from '@storybook/react/types-6-0';

import { CartIcon, CartIconProps } from './CartIcon';

export default {
  title: 'Atoms/CartIcon',
  component: CartIcon,
  args: {
    quantity: 1,
  },
} as Meta<CartIconProps>;

export const Default: Story = args => <CartIcon {...args} />;
