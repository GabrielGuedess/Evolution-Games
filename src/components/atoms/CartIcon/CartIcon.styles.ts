import media from 'styled-media-query';

import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    position: relative;
    width: fit-content;
    transition: color 0.3s linear;

    :is(:hover, :focus) {
      cursor: pointer;
      color: ${theme.colors.primary};
    }
  `}
`;

export const Badge = styled.span`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    background: ${theme.colors.primary};
    box-shadow: ${theme.shadows.text};
    color: ${theme.colors.white};
    font-size: 1.2rem;
    line-height: 1rem;
    border-radius: 1rem;
    padding: 0.3rem 0.6rem;
    position: absolute;
    top: -0.4rem;
    right: -0.4rem;
    cursor: pointer;

    ${media.greaterThan('medium')`
      font-size: 0.8rem;
    `}
  `}
`;
