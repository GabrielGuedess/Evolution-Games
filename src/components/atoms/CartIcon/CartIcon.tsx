import * as AccessibleIcon from '@radix-ui/react-accessible-icon';
import { HandbagSimple } from 'phosphor-react';

import * as S from './CartIcon.styles';

export type CartIconProps = {
  quantity?: number;
};

export const CartIcon = ({ quantity }: CartIconProps) => (
  <S.Wrapper>
    <AccessibleIcon.Root label="Shopping Bag">
      <HandbagSimple size={24} aria-label="Cart Icon" />
    </AccessibleIcon.Root>

    {!!quantity && quantity > 0 && (
      <S.Badge role="img" aria-label="Cart Items">
        {quantity}
      </S.Badge>
    )}
  </S.Wrapper>
);
