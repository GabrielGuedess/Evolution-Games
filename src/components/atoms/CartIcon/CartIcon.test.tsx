import { renderWithProviders } from 'utils/tests/helpers';

import { screen } from '@testing-library/react';

import { CartIcon } from './CartIcon';

describe('<CartIcon />', () => {
  it('should render without quantity', () => {
    // Arrange
    const { container } = renderWithProviders(<CartIcon />);

    // Assert
    expect(screen.getByLabelText('Cart Icon')).toBeInTheDocument();
    expect(screen.queryByLabelText('Cart Items')).not.toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render without quantity', () => {
    renderWithProviders(<CartIcon quantity={1} />);

    // Assert
    expect(screen.getByLabelText('Cart Icon')).toBeInTheDocument();
    expect(screen.getByLabelText('Cart Items')).toBeInTheDocument();
  });
});
