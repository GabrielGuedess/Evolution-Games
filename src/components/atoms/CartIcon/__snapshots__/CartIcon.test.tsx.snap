// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<CartIcon /> should render without quantity 1'] = `
.c0 {
  position: relative;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  -webkit-transition: color 0.3s linear;
  transition: color 0.3s linear;
}

.c0:is(:hover,:focus) {
  cursor: pointer;
  color: #855aff;
}

<div
  class="c0"
>
  <svg
    aria-hidden="true"
    aria-label="Cart Icon"
    fill="currentColor"
    focusable="false"
    height="24"
    viewBox="0 0 256 256"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="none"
      height="256"
      width="256"
    />
    <path
      d="M208.8,72H47.2a8.1,8.1,0,0,0-8,7.1L25,207.1a8,8,0,0,0,7.9,8.9H223.1a8,8,0,0,0,7.9-8.9l-14.2-128A8.1,8.1,0,0,0,208.8,72Z"
      fill="none"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="16"
    />
    <path
      d="M88,72a40,40,0,0,1,80,0"
      fill="none"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="16"
    />
  </svg>
  <span
    style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
  >
    Shopping Bag
  </span>
</div>
`;
